app/models/manageiq/providers/vmware/infra_manager/ovf_service.rb

```diff
@@ -0,0 +1,150 @@
+class ManageIQ::Providers::Vmware::InfraManager::OvfService < ServiceGeneric
+  delegate :template, :manager, :to => :service_template, :allow_nil => true
+
+  # A chance for taking options from automate script to override options from a service dialog
+  def preprocess(action, new_options = {})
+    return unless action == ResourceAction::PROVISION
+
+    if new_options.present?
+      _log.info("Override with new options:")
+      $log.log_hashes(new_options)
+    end
+
+    save_action_options(action, new_options)
+  end
+
+  def execute(action)
+    return unless action == ResourceAction::PROVISION
+
+    library_item_deploy_queue
+  end
+
+  def library_item_deploy_queue
+    task_options = {:action => "Deploying VMware Content Library Item", :userid => "system"}
+    queue_options = {
+      :class_name  => self.class.name,
+      :instance_id => id,
+      :method_name => "library_item_deploy",
+      :args        => {},
+      :role        => "ems_operations",
+      #:queue_name  => manager.queue_name_for_ems_operations,
+      :zone        => manager.my_zone
+    }
+
+    task_id = MiqTask.generic_action_with_callback(task_options, queue_options)
+    task = MiqTask.wait_for_taskid(task_id)
```

Comment:
I don't know much about how the Service provision side works but should we be waiting for the task to finish in a separate state?

Reply:
The wait here is to make sure that the `library_item_deploy` is done when automate moves from `execute` to next step. Otherwise, when automate moves to next step `check_completed`, `library_item_deploy` might still be in the queue and not being executed yet.

Reply:
This will lock-up a worker while the content library is being deployed?

Reply:
I think so. 
I can remove the wait here then rely on `check_completed` to do the work right.

Reply:
I think that makes more sense

Reply:
Hi @Lucy, @agrare I agree that we should remove the wait and rely on ```check_completed```.